cpp/CMakeLists.txt

```diff
@@ -629,19 +594,31 @@ if(compiled IN_LIST raft_FIND_COMPONENTS)
 endif()
 ]=]
 )
+set(raft_components compiled distributed)
+set(raft_export_sets raft-compiled-exports raft-distributed-exports)
+if(TARGET raft_lib)
+  list(APPEND raft_components compiled)
+  list(APPEND raft_export_sets raft-compiled-lib-exports)
+endif()
 
-# Use `rapids_export` for 22.04 as it will have COMPONENT support
-include(cmake/modules/raft_export.cmake)
-raft_export(
-  INSTALL raft COMPONENTS compiled distributed EXPORT_SET raft-exports GLOBAL_TARGETS raft compiled
-  distributed NAMESPACE raft:: DOCUMENTATION doc_string FINAL_CODE_BLOCK code_string
+rapids_export(
+  INSTALL raft
+  EXPORT_SET raft-exports
+  COMPONENTS ${raft_components}
+  COMPONENTS_EXPORT_SET ${raft_export_sets}
+  GLOBAL_TARGETS raft compiled distributed
```

Comment:
Making distributed a build time flag would mean we would need 2 conda builds to make packages. If the goal is to have everything generated in a single conda build we can't have components gated behind CMake options.

> Could we just always create the raft::compiled target so that it'll never actually build anything if a user never links against it?

So would that mean we never have a version of raft on conda that doesn't have pre-compiled libraries?

Reply:
> So would that mean we never have a version of raft on conda that doesn't have pre-compiled libraries?

I think this would directly go against the goals of #1386